lapce-ui/src/logging.rs

```diff
@@ -0,0 +1,87 @@
+use fern::Dispatch;
+use log::LevelFilter;
+
+fn apply(
+    dispatch: Dispatch, module: Option<&str>, level: LevelFilter
+) -> Dispatch {
+    if let Some(module) = module {
+        dispatch.level_for(module.to_string(), level)
+    } else {
+        dispatch.level(level)
+    }
+}
+
+fn set_level(
+    dispatch: Dispatch, module: Option<&str>, level: &str
+) -> Dispatch {
+    use LevelFilter::*;
+    match level {
+        "off" => apply(dispatch, module, Off),
+        "error" => apply(dispatch, module, Error),
+        "warn" => apply(dispatch, module, Warn),
+        "info" => apply(dispatch, module, Info),
+        "debug" => apply(dispatch, module, Debug),
+        "trace" => apply(dispatch, module, Trace),
+        val @ _ => {
```

Comment:
This is an interesting pattern. Why did you chose this over a bare identifier, like `other => ...`?

Reply:
> This is an interesting pattern. Why did you chose this over a bare identifier, like `other => ...`?

Because I forgot that a mere `other => ...` will bind anything else. :P

I will fix it.